File: src/main/scala/avalon/api/CustomGroupResponder.scala
package avalon.api

import java.util.regex.Pattern

import avalon.util.{GroupConfig, GroupMessage}
import org.apache.commons.lang3.builder.{EqualsBuilder, HashCodeBuilder}

trait CustomGroupResponder {
	def doPost(message: GroupMessage, groupConfig: GroupConfig): Unit

	def getHelpMessages: List[String]

	def getKeyWordRegex: Pattern

	override def equals(obj: scala.Any): Boolean = {
		if (obj == null || (!obj.isInstanceOf[CustomGroupResponder])) return false
		val that = obj.asInstanceOf[CustomGroupResponder]
		new EqualsBuilder()
			.append(getKeyWordRegex, that.getKeyWordRegex)
			.isEquals
	}

	override def hashCode(): Int =
		new HashCodeBuilder(17, 37).append(getKeyWordRegex).hashCode()
}

File: src/main/kotlin/avalon/api/CustomGroupResponder.kt
package avalon.api

import avalon.util.GroupConfig
import avalon.util.GroupMessage
import org.apache.commons.lang3.builder.EqualsBuilder
import org.apache.commons.lang3.builder.HashCodeBuilder
import java.util.regex.Pattern

abstract class CustomGroupResponder {
	abstract fun doPost(message: GroupMessage, groupConfig: GroupConfig)

	abstract fun getHelpMessage(): String

	abstract fun getKeyWordRegex(): Pattern

	override fun hashCode() = HashCodeBuilder(17, 37).toHashCode()
	override fun equals(other: Any?): Boolean {
		if (this === other) return true
		if (javaClass != other?.javaClass) return false
		other as CustomGroupResponder
		return EqualsBuilder().append(other.getKeyWordRegex().toString(), getKeyWordRegex().toString()).isEquals
	}
}